File: src/ts/magma/app/compile/type/Placeholder.ts
import { Definition } from "../../../../magma/app/compile/define/Definition";
import { Option } from "../../../../magma/api/option/Option";
import { None } from "../../../../magma/api/option/None";
import { ValueCompiler } from "../../../../magma/app/ValueCompiler";
export class Placeholder {
	input: string;
	constructor (input: string) {
		this.input = input;
	}
	static generatePlaceholder(input: string): string {
		let replaced = input.replace("/*", "start").replace("*/", "end")/*unknown*/;
		return "/*" + replaced + "*/"/*unknown*/;
	}
	generateNode(): string {
		return Placeholder.generatePlaceholder(this.input)/*unknown*/;
	}
	generate(): string {
		return Placeholder.generatePlaceholder(this.input)/*unknown*/;
	}
	asDefinition(): Option<Definition> {
		return new None<Definition>()/*unknown*/;
	}
	generateBeforeName(): string {
		return ""/*unknown*/;
	}
	generateSimple(): string {
		return ValueCompiler.generateValue(this)/*unknown*/;
	}
	is(type: string): boolean {
		return false/*unknown*/;
	}
}

File: src/java/magma/app/compile/type/Placeholder.java
package magma.app.compile.type;

import magma.api.option.None;
import magma.api.option.Option;
import magma.app.ValueCompiler;
import magma.app.compile.define.Definition;
import magma.app.compile.define.Parameter;

public record Placeholder(String input) implements Parameter, Type {
    public static String generatePlaceholder(String input) {
        var replaced = input
                .replace("/*", "start")
                .replace("*/", "end");

        return "/*" + replaced + "*/";
    }

    public String generateType() {
        return Placeholder.generatePlaceholder(this.input);
    }

    @Override
    public String generate() {
        return Placeholder.generatePlaceholder(this.input);
    }

    @Override
    public Option<Definition> asDefinition() {
        return new None<Definition>();
    }

    public String generateBeforeName() {
        return "";
    }

    public String generateSimple() {
        return ValueCompiler.generateValue(this);
    }

    public boolean is(String type) {
        return false;
    }
}
